src/stream/src/executor/managed_state/dynamic_filter.rs

```diff
@@ -154,8 +155,14 @@ impl<S: StateStore> RangeCache<S> {
 
         let own_vnodes = self.vnodes.ones().collect_vec();
         for pk_range in missing_ranges {
-            for vnode in &own_vnodes {
-                self.add_vnode_range(*vnode, &pk_range).await?;
+            for (vnode, mut map) in try_join_all(
```

Comment:
As discovered by @chenzl25, `join_all` with too many futures actually degrades the performance if there's not too much data (or hit the cache in majority), as it'll brings more scheduling overhead. We may consider set a maximum concurrent degree with `buffer_unordered`.

BTW, we may also fix this for storage table.

Reply:
> As discovered by @chenzl25, `join_all` with too many futures actually degrades the performance if there's not too much data (or hit the cache in majority), as it'll brings more scheduling overhead. We may consider set a maximum concurrent degree with `buffer_unordered`.
> 
> BTW, we may also fix this for storage table.

fixed